main.tf

```diff
@@ -39,3 +40,10 @@ resource "azurerm_servicebus_queue_authorization_rule" "listen_auth_rule" {
 
   listen = true
 }
+
+resource "azurerm_servicebus_subscription_rule" "BBA3-filter" {
```

Comment:
You can't do this here as this module is meant to be generic, not just for SSCS. You need to make this generic (see how it's done in servicebus_queue with vars)